test-support/lit-exec-qemu.py

```diff
@@ -15,6 +15,27 @@
 # This script requires Python 3.6 or later
 assert sys.version_info >= (3, 6)
 
+picolibc_binary_name = "check-picolibc"
```

Comment:
llvm tests have lit instructions in *.c and *.cpp files thus it is easy for them to use "not" tool for specific test. In this implementation we use template for all tests and it is more complicated. In my opinion one solution would be to keep a mapping in CMake file for each picolibc test name and explictly indicate to use "not". Or by doing this in python wrapper as i did.

Reply:
Another option is to use a different template for each test. It will be more repetitive, but if it makes things simpler then it could be worthwhile.

If you want to go the route of adding more logic in CMake then I suggest moving the picolibc test CMake code into a [subdirectory](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) because the main CMakeLists.txt is quite bloated at this point.